src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -3931,12 +3959,12 @@ public TElement LastOrDefault()
 
         public TElement[] ToArray()
         {
-            return new TElement[0]; // consider replacing with Array.Empty<TElement>()
+            return Array.Empty<TElement>();
```

Comment:
If the decision to use `Array.Empty` in these cases had been made before I created this class, I probably would have used its enumerator for the `GetEnumerator()` of this class and not bothered with its implementing `IEnumerator`. Not sure whether it's worth changing to that now, though.